develop/nix/devshell.nix

```diff
@@ -0,0 +1,64 @@
+{ inputs
+, ...
+}:
+
+{
+  perSystem = { pkgs, lib, ... }: {
+    devshells.default =
+      let
+        rust-toolchain = with pkgs;
+          [
+            ((rust-bin.fromRustupToolchainFile ../../rust-toolchain).override {
+              extensions = [ "rust-src" "rust-analyzer" ];
+            })
+          ] ++ [
+            cargo-make
+            cargo-nextest
+            cargo-binstall
+            cargo-sort
+            typos
+          ];
+      in
+      {
+        imports = [
+          "${inputs.devshell}/extra/language/rust.nix"
+        ];
+        language.rust.enableDefaultToolchain = false;
+        packages = rust-toolchain ++ (with pkgs; [
+          gcc
+          lld
+          protobuf
+          pkg-config
+          cyrus_sasl.out
+
+          gnumake
+          cmake
+          maven
+          jdk17_headless
+
+          tmux
+          postgresql
+          patchelf
+        ]);
```

Comment:
Better to add some comment here to remind future maintainers that these pkgs are listed in `developer-guide.md`'s "Set up the development environment" subsection. Always point to the source of trust.